e3sm_to_cmip/cmor_handlers/handlers.yaml

```diff
@@ -654,14 +662,6 @@
   formula: FSDSC - FSNSC
   positive: up
   levels: null
-- name: rsut
-  units: W m-2
-  raw_variables: [SOLIN, FSNTOA]
-  table: CMIP6_Amon.json
-  unit_conversion: null
-  formula: SOLIN - FSNTOA
-  positive: up
-  levels: null
```

Comment:
`rsut` was added through this recent PR by @chengzhuzhang in Sep/2023.

https://github.com/E3SM-Project/e3sm_to_cmip/pull/210

> FSUTOA and FSUTOAC are used in e3sm_to_cmip for directly getting rsut and rsutcs. For v3 output, both FSUTOA and FSUTOAC are removed. This PR is to update formula to use available variables, i.e., rsut = SOLIN - FSNTOA, and rsutcs = SOLIN - FSNTOAC

Not sure if it should be removed?

Reply:
@TonyB9000 comment on the reason for this change: https://github.com/E3SM-Project/e3sm_to_cmip/issues/251#issuecomment-2013025956

Reply:
The issue here is that, for the very same table (CMIP6_Amon.json), there are two distinct formulas for computing the variable:

```
- CMIP6 Name: rsut
  CMIP6 Table: CMIP6_Amon.json
  CMIP6 Units: W m-2
  E3SM Variables: SOLIN, FSNTOA
```
and
```
- CMIP6 Name: rsut
  CMIP6 Table: CMIP6_Amon.json
  CMIP6 Units: W m-2
  E3SM Variables: FSUTOA		
```
If you need to generate Amon.rsut, which native variables will you select and why?

If model version (v1 vs v2 vs v3) is a criterion, it needs to be an attribute of the handler.

If both formulas are required, we cannot use only the "E3SM variables" to distinguish them, since we rely upon these handlers to know which E3SM variables to seek for a given CMIP6 variable (eg rsut).  We cannot reasonably proceed by waiting until the native data is examined to see which of several combinations of variable may be appropriate.  And if more than one formula could be satisfied, does one toss a coin to decide which to apply?

I used https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/925304036/Amon+variable+conversion+table as my guide in which to keep.  It listed only the latter formula.

Reply:
I somehow missed these comments. We should not remove these additions in order to support v3 output. Because 
```
FSUTOA and FSUTOAC are used in e3sm_to_cmip for directly getting rsut and rsutcs. For v3 output, both FSUTOA and FSUTOAC are removed. This PR is to update formula to use available variables, i.e., rsut = SOLIN - FSNTOA, and rsutcs = SOLIN - FSNTOAC
```

Reply:
@chengzhuzhang OK, but we have a problem if there is no way to know which formula to select.

`e3sm_to_cmip --info mode` queries the handlers.  If it is asked "What native vars are needed to generate Amon.rsut", and the answer is "Either (FSUTOA) or (SOLIN, FSNTOA), take your pick", then we have a problem.  How does e2c know which handler to use?  (v2 versus v3 of course, but where is this determined?)

Reply:
The process of how handlers are selected  is  documented here:https://github.com/E3SM-Project/e3sm_to_cmip/blob/0f91fb4abc839e2ec04fe8e770ad7308a41c6620/docs/source/var-handlers.rst#how-e3sm_to_cmip-derives-atmosphereland-handler. This is a process to loop over handlers of a same variable is involved.

Reply:
Yes, I see now.  It says:

> derives the appropriate variable handlers to use **based on the available E3SM variables in the input datasets**.

What this means is that we cannot use "info mode 1" or "info mode 2" to select the native variables to employ for a given CMIP6 variable, (when doing "variable at a time") since they provide no "input datasets".  We MUST supply a raw dataset as well (mode 3).

ASIDE:  What would happen in the native data had variables that satisfied more than one formula (handler)?  Seem the outcome would be indeterminate.

I will have to conduct thorough "info mode 3" testing, to see if ambiguous returns still occur.